src/main/java/org/concordion/internal/listener/ResourcesFactory.java

```diff
@@ -0,0 +1,191 @@
+package org.concordion.internal.listener;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.concordion.api.Resources;
+import org.concordion.api.Resources.InsertType;
+import org.concordion.api.Fixture;
+import org.concordion.api.Resource;
+import org.concordion.api.extension.ConcordionExtender;
+import org.concordion.internal.util.IOUtil;
+
+/**
+ * Copy resources to the destination when specification is built.  Resources are copied to the same relative path as they are in the project.
+ * Currently only supports resources on the file path.  Support for resources in Jars may come at a later date but will require java 1.7.
+ * 
+ * @author sumnera
+ */
+public class ResourcesFactory {
+	private boolean includeDefaultStyling = true;
+	private List<ResourceToCopy> sourceFiles = new ArrayList<ResourceToCopy>();
+		
+	public boolean includeDefaultStyling() {
+		return this.includeDefaultStyling;
+	}
+	
+	public List<ResourceToCopy> getFiles() {
+		return this.sourceFiles;
+	}
+	
+	public ResourcesFactory(ConcordionExtender builder, Fixture fixture) {
+		
+		File root = getRootPath(fixture.getFixtureClass());
+		List<Class<?>> classes = getClassHierarchyParentFirst(fixture.getFixtureClass());
+		
+		for (Class<?> class1 : classes) {
+			if (class1.isAnnotationPresent(Resources.class)) {
+	            Resources annotation = class1.getAnnotation(Resources.class);
+	            
+	            if (!annotation.includeDefaultStyling()) {
+    				includeDefaultStyling = false;
+    			}
+    	
+	            sourceFiles.addAll(getResourcesToAdd(class1, annotation, root));
+	        }
+        }
+		
```

Comment:
I'd suggest having this class return the list of resources to copy to ConcordionBuilder, and moving the following block of code into ConcordionBuilder so that we are not calling back into ConcordionBuilder.